### PR TITLE
typedIdent no longer destroys attachments

### PR DESCRIPTION
**Pull request by xeno-by**
When transforming Idents to qualified Selects, typedIdent used to forget
about carrying original attachments to the resulting tree. Not anymore.

### CODE REVIEW

**Comment by xeno-by** [comment]
review @retronym

**Comment by paulp** [comment]
We should place a moratorium on Unit-returning methods. What is the advantage of a Unit-returning setter as compared to

```
def setAttachments...: this.type = ...
```

If you have that the diff to Typers is a single line rather than a page.

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 1]
```diff
@@ -5147,27 +5147,31 @@ trait Typers extends Modes with Adaptations with Tags {
...
+            }
+            // Inferring classOf type parameter from expected type.  Otherwise an
+            // actual call to the stubbed classOf method is generated, returning null.
+            else if (isPredefMemberNamed(defSym, nme.classOf) && pt.typeSymbol == ClassClass && pt.typeArgs.nonEmpty)
+              typedClassOf(tree, TypeTree(pt.typeArgs.head))
+            else {
+              val tree1 = (
+                if (qual == EmptyTree) tree
+                // atPos necessary because qualifier might come from startContext
+                else atPos(tree.pos)(Select(qual, name))
```
Comment (retronym) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala]: Why not `treeCopy.Select(tree, qual, name)` to get the position and attachments in one fell swoop?

**Comment by xeno-by** [comment]
Addressed the feedback

**Comment by xeno-by** [comment]
PLS REBUILD ALL

**Comment by paulp** [comment]
lgtm too.

**Pull request merged**